### PR TITLE
Print the version number without serializing the state

### PR DESCRIPTION
**Pull request by kalikiana**
See: [poo#75232](https://progress.opensuse.org/issues/75232)

### CODE REVIEW

**Comment by okurz** [comment]
As visible in https://github.com/os-autoinst/os-autoinst/pull/1611/checks?check_run_id=1656848742#step:3:572 the test already takes 48s:


```
6: [14:27:21] ./14-isotovideo.t ........................ ok    48792 ms ( 0.02 usr  0.00 sys + 45.07 cusr  2.20 csys = 47.29 CPU)
```

can you please check that you have not introduced a significant increase of runtime?

**Comment by kalikiana** [comment]
> can you please check that you have not introduced a significant increase of runtime?

The new subtest adds around `500ms` which is basically the time it takes for `isotovideo` to execute anything.

But apparently the test in general could be faster, so I optimized the log handling a bit.

File: t/14-isotovideo.t [hunk 1]
```diff
@@ -22,27 +23,28 @@ my $pool_dir     = "$dir/pool";
...
+    my @cmd = ($^X, "$toplevel_dir/isotovideo", '-d', $args{default_opts}, split(' ', $args{opts}));
+    note("Starting isotovideo with: @cmd");
+    capture [$args{exit_code}], @cmd;
 }
 
+subtest 'get the version number' => sub {
+    # Make sure we're in a folder we can't write to, no base_state.json should be created here
+    chdir('/');
+    combined_like { system $^X, "$toplevel_dir/isotovideo", '--version' } qr/Current version is.+\[interface v20\]/, 'version printed';
+    ok(!-e bmwqemu::STATE_FILE, 'no state file was written');
```
Comment (Amrysliu) [inline t/14-isotovideo.t:41]: I do not understand this test code clearly. is it used to prove that we don't write any message to `STATE_FILE ` file?  I guess we also need to check $fatal_error will not be filed with messages when running `--version`

**Review by Amrysliu — COMMENTED** [review]
The changes of `$SIG{__DLE__}` looks good to me. Please check the CI check, it fails at 14-isotovide.t

File: t/14-isotovideo.t (hunk 1 above)
Comment (kalikiana) [reply t/14-isotovideo.t:41]: Yes. If `$fatal_error` is set, the file is written. So I'm checking that it's not written. Which means there's no error.

**Review by Amrysliu — APPROVED** [review]
LGTM

**Comment by Martchus** [comment]
Thanks, this was really annoying. Considering my local worker log it seems like it worked :+1:

**Pull request merged**